app/server.js

```diff
@@ -1,31 +1,179 @@
-const express = require('express');
-const bodyParser = require('body-parser');
-const jwt = require('jsonwebtoken');
-const swaggerUi = require('swagger-ui-express');
-const YAML = require('yamljs');
-const swaggerDocument = YAML.load('./swagger.yaml'); // Replace './swagger.yaml' with the path to your Swagger file
+const express = require("express");
+const bodyParser = require("body-parser");
+const jwt = require("jsonwebtoken");
+const swaggerUi = require("swagger-ui-express");
+const YAML = require("yamljs");
+const swaggerDocument = YAML.load("./swagger.yaml");
 const app = express();
 
 app.use(bodyParser.json());
 
 // Importing the data from JSON files
-const users = require('../initial-data/users.json');
-const brands = require('../initial-data/brands.json');
-const products = require('../initial-data/products.json');
+const users = require("../initial-data/users.json");
+const brands = require("../initial-data/brands.json");
+const products = require("../initial-data/products.json");
 
 // Error handling
 app.use((err, req, res, next) => {
-	console.error(err.stack);
-	res.status(500).send('Something broke!');
+  console.error(err.stack);
+  res.status(500).send("Something broke!");
 });
 
 // Swagger
-app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
+app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swaggerDocument));
+
+app.get("/brands", (request, response) => {
+  response.send(brands);
+});
+
+app.get("/products", (request, response) => {
+  response.send(products);
+});
+
+app.get("/brands/:id/products", (request, response) => {
+  // Get all products that match the brand id
+  // The sample data provided has a categoryId property that aligns with the brand id numbers - I think it would be better named "brandId" but chose not to change the existing data
+  const filteredProducts = products.filter((product) => {
+    return product.categoryId == request.params.id;
+  });
+
+  if (filteredProducts.length === 0) {
+    response.status(404);
+    response.send({ message: "No products found" });
+  } else {
```

Comment:
not accurate, is not a client error that there are no filtered products, you can return an empty array and show an empty state in the client.